gitpod-io/gitpod
# [dev] sweeper: watch for branch instead of activity

## Description
Extends `sweeper` so it can check whether the **branch it's associated with got deleted.**

Also reconfigures the `sweeper` to delete a preview env if **either**:
 - it's branch got deleted, OR
 - there has been no activity in the preview env for `2d`

[link](https://github.com/gitpod-io/gitpod/blob/3a7de9af420bcc27ce8951fcf7fe07773fc8d12e/.werft/build.ts#L463-L469)

## Motivation
For me, it's becoming more and more of a burden to prepare a running preview environment for reviewers/as a reviewer because:
 - builds take time
 - builds are flaky
 - I have to re-do everything when I got pulled out of context (taken down after 3h of inactivity as per current `sweeper` config)
 - I have no clue when a reviewer finds time to actually review

When `sweeper` was conceived it was meant to work under the following conditions:
 - technical problems with too many parallel preview envs: purge old ones
 - relative high costs incurred by too many preview envs: purge old ones
 - a super-local team that was "always in sync"

All of the statements are not longer true anymore.

Furthermore, we want to increase cycles speeds, and have a bit of money to spend on this, so let's unblock ourselves. :slightly_smiling_face: 

WDYT?

## Related Issue(s)

Fixes #

## How to test

- create a branch from this branch
- wait for the preview environment
- `kubectl logs sweeper-` to check the output
- delete your branch on GitHub
- wait up `10m` until the preview env gets purged

## Release Notes

```release-note
NONE
```

## Documentation
